provider/redis.go

```diff
@@ -95,48 +115,138 @@ func (store *redisOnlineStore) DeleteTable(feature, variant string) error {
+		} else {
+			value = "0"
+		}
+	case time.Time:
+		value = v.Format(time.RFC3339)
+	// Vector parsed from parquet files will have the type map[string]interface{}
+	// and require further processing to be stored in Redis as a string.
+	case map[string]interface{}:
+		serialized, err := table.formatVector(v)
+		if err != nil {
+			return err
+		}
+		value = serialized
+	default:
+		return fmt.Errorf("type %T of value %v is unsupported", value, value)
+	}
+	cmd := client.B().
+		Hset().
+		Key(table.key.String()).
+		FieldValue().
+		FieldValue(entity, value.(string)).
+		Build()
+	res := client.Do(context.TODO(), cmd)
+	if res.Error() != nil {
+		return res.Error()
 	}
 	return nil
 }
 
 func (table redisOnlineTable) Get(entity string) (interface{}, error) {
-	val := table.client.HGet(context.TODO(), table.key.String(), entity)
-	if val.Err() != nil {
+	client := *table.client
+	cmd := client.B().
+		Hget().
+		Key(table.key.String()).
+		Field(entity).
+		Build()
+	resp := client.Do(context.TODO(), cmd)
+	if resp.Error() != nil {
 		return nil, &EntityNotFound{entity}
 	}
 	var result interface{}
 	var err error
+	val, err := resp.ToString()
+	if err != nil {
+		return nil, err
+	}
 	switch table.valueType {
 	case NilType, String:
-		result, err = val.Result()
+		result, err = val, nil
 	case Int:
-		result, err = val.Int()
+		result, err = strconv.Atoi(val)
 	case Int32:
-		if res, err := val.Int(); err == nil {
-			result = int32(res)
+		if result, err = strconv.ParseInt(val, 10, 32); err == nil {
+			result = int32(result.(int64))
 		}
 	case Int64:
-		result, err = val.Int64()
+		result, err = strconv.ParseInt(val, 10, 64)
 	case Float32:
-		result, err = val.Float32()
+		if result, err = strconv.ParseFloat(val, 32); err == nil {
+			result = float32(result.(float64))
+		}
 	case Float64:
-		result, err = val.Float64()
+		result, err = strconv.ParseFloat(val, 64)
 	case Bool:
-		result, err = val.Bool()
+		result, err = strconv.ParseBool(val)
 	case Timestamp, Datetime: // Maintains compatibility with previously create timestamp tables
-		var t time.Time
-		t, err = val.Time()
-		if err == nil {
-			result = t.Local()
-		}
+		// maintains compatibility with go-redis implementation:
+		// https://github.com/redis/go-redis/blob/v8.11.5/command.go#L939
+		result, err = time.Parse(time.RFC3339Nano, val)
+	case Vector32:
+		result, err = rueidis.ToVector32(val), nil
 	default:
-		result, err = val.Result()
+		result, err = val, nil
 	}
 	if err != nil {
-		return nil, fmt.Errorf("could not cast value: %s to %s: %w", val.String(), table.valueType, err)
+		return nil, fmt.Errorf("could not cast value: %v to %s: %w", resp, table.valueType, err)
 	}
 	return result, nil
 }
+
+// formatVector takes a map[string]interface{} and returns a string representation of the vector
+// using the VectorString32 method provided by the rueidis Redis package.
+func (table redisOnlineTable) formatVector(value map[string]interface{}) (string, error) {
```

Comment:
I was wondering this, too, especially because the test I wrote for the parquet iterator is nearly identical to the logic here. The issue was that I mistyped the unit test in `redis_test.go`, which led to a failing test even though the E2E flow worked as expected.

I can make this change and have it return `interface{}` with the underlying type being `[]float32`. Does this work for you?